File: omnipod-dash/src/main/java/info/nightscout/androidaps/plugins/pump/omnipod/dash/driver/comm/exceptions/ScanFailFoundTooManyException.java
package info.nightscout.androidaps.plugins.pump.omnipod.dash.driver.comm.exceptions;

import java.util.ArrayList;
import java.util.Collections;
import java.util.List;

import info.nightscout.androidaps.plugins.pump.omnipod.dash.driver.comm.scan.BleDiscoveredDevice;

public class ScanFailFoundTooManyException extends ScanFailException {
    private final List<BleDiscoveredDevice> devices;

    public ScanFailFoundTooManyException(List<BleDiscoveredDevice> devices) {
        super();
        this.devices = new ArrayList<>(devices);
    }

    public List<BleDiscoveredDevice> getDiscoveredDevices() {
        return Collections.unmodifiableList(this.devices);
    }
}

File: omnipod-dash/src/main/java/info/nightscout/androidaps/plugins/pump/omnipod/dash/driver/comm/exceptions/ScanFailFoundTooManyException.kt
package info.nightscout.androidaps.plugins.pump.omnipod.dash.driver.comm.exceptions

import info.nightscout.androidaps.plugins.pump.omnipod.dash.driver.comm.scan.BleDiscoveredDevice
import java.util.*

class ScanFailFoundTooManyException(devices: List<BleDiscoveredDevice>) : ScanException("Found more than one Pod") {

    private val devices: List<BleDiscoveredDevice> = ArrayList(devices)
    val discoveredDevices: List<BleDiscoveredDevice>
        get() = Collections.unmodifiableList(devices)
}
